### PR TITLE
Update filter_query module to accommodate multiple queries 

### PR DESCRIPTION
**Pull request by kylacochrane**
This PR resolves an issue with the `filter_query` process and workflow, where the final CSV file (`new_addresses.csv`) was being overwritten. This file is essential for the nf-iridanext plugin to generate the JSON file for IRIDA Next with the correct metadata.

Key updates in this PR include:

1. Modifying the workflow and process to support multiple queries.
2. Adding tests to main.nf.test to verify the pipeline's capability to handle multiple queries.
3. Updating the input_check process to include `fair true` to maintain the order of query samples.

## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).

### CODE REVIEW

**Review by apetkau — APPROVED** [review]
Thanks so much @kylacochrane . This looks great 😄

File: tests/pipelines/main.nf.test [hunk 1]
```diff
@@ -49,6 +54,61 @@ nextflow_pipeline {
...
+            def expected_profile_query_tsv = path("$baseDir/tests/data/profiles/expected-profile_queries2.tsv")
+            assert actual_profile_query.text == expected_profile_query_tsv.text
+
+            // Check computed pairwise distances
+            def actual_distances = path("$launchDir/results/distances/results.text")
+            def expected_distances = path("$baseDir/tests/data/distances/expected_pairwise_queries_dists.txt")
+            assert actual_distances.text == expected_distances.text
+
+            // Verify cluster file
+            def actual_cluster = path("$launchDir/results/cluster/expected_clusters.txt")
```
Comment (emarinier) [inline tests/pipelines/main.nf.test:87]: Is it the case that the output filename for the clusters is `expected_clusters.txt`?

**Review by emarinier — APPROVED** [review]
Looks good to me. Just one question on a filename in a test.

File: tests/pipelines/main.nf.test (hunk 1 above)
Comment (kylacochrane) [reply tests/pipelines/main.nf.test:87]: Yes it is :)

Comment (kylacochrane) [reply tests/pipelines/main.nf.test]: Although ... it is a little confusing to call the output file for the reference clusters this way.
I have changed the output filename to: `reference_clusters` in the `cluster_file` process and updated the tests.

See changes here: [15b7090](https://github.com/phac-nml/gasnomenclature/pull/14/commits/15b7090691f90dbb969be9db01218715673201ca)

**Pull request merged**